#VRML_SIM R2023b utf8

EXTERNPROTO "../protos/FireSmoke.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/trees/protos/Forest.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/dji/mavic/protos/Mavic2Pro.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/factory/manhole/protos/SquareManhole.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/road/protos/StraightRoadSegment.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

DEF World WorldInfo {
  title "Wildfire scenario"
  basicTimeStep 8
  defaultDamping Damping {
    linear 0.5
    angular 0.5
  }
}
Viewpoint {
  orientation -0.17152773171417718 0.1715277317141773 0.9701321943456874 1.6011146149123783
  position 0.003835383610936989 -50.86939457776317 1.4569763670330347
  near 0.2
  follow "Mavic 2 PRO"
  followType "Mounted Shot"
}
TexturedBackground {
  texture "noon_park_empty"
}
TexturedBackgroundLight {
  texture "noon_park_empty"
}
DEF FlightArea Floor {
  translation 0 0 -0.0001
  rotation 0 0 1 1.5707963267948966
  size 100 100
  tileSize 1 1
  appearance PBRAppearance {
    baseColorMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/default/worlds/textures/dry_grass.jpg"
      ]
    }
    roughness 1
    metalness 0
  }
}
DEF ForestArea Forest {
  rotation 0 0 1 -1.5707963267948966
  shape [
    -50 -50
    40 -50
    40 50
    -50 50
  ]
  density 0.5
  randomSeed 2
  maxHeight 10
  minHeight 3
}
Solid {
  translation 0 5 5
  name "ForestBB"
  boundingObject Box {
    size 100 90 10
  }
}
StraightRoadSegment {
  translation 50 -45.7 0
  rotation 0 0 1 3.141592653589793
  length 100
}
SquareManhole {
  translation 0 -47 0.02
  rotation 0 0 1 -1.5707926535897934
  name "Starting point"
  size 1 1 0.05
}
Solid {
  translation 0 -47.12 0.052
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/default/worlds/textures/lightwood.jpg"
          ]
        }
        roughness 0.6
        metalness 0
      }
      geometry DEF NIVEL Box {
        size 0.09 0.1 0.013301
      }
    }
  ]
  boundingObject USE NIVEL
  physics Physics {
    density 500
  }
}
DEF FireSmoke FireSmoke {
  translation 0 -40 0.5
  fireHeight 12.769389698745128
  fireRadius 1.5181178932102077
}
DEF Drone Mavic2Pro {
  translation 0 -47 0.117
  rotation 0 0 1 1.5708
  controller "drone_controller"
  bodySlot [
    Receiver {
      name "ActionReceiver"
      channel 6
    }
    Emitter {
      name "StateEmitter"
      channel 4
    }
    DistanceSensor {
      translation 0.06 0.0332 0.003
      rotation 1 0 0 1.5708
      name "front left dist sonar"
      lookupTable [
        0.5 50 0
        20 2000 0
        40 4000 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation 0.06 -0.033 0.003
      rotation 1 0 0 1.5708
      name "front right dist sonar"
      lookupTable [
        0.5 50 0
        20 2000 0
        40 4000 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.1364 0 -0.011
      rotation 0 -1 0 3.14159
      children [
        DEF DIST_SENSOR Pose {
          rotation 0 0 -0.9999999999999999 1.5707953071795862
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.5 0.5 0.5
                roughness 0.8
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.0035
              }
            }
          ]
        }
      ]
      name "rear top dist sonar"
      lookupTable [
        0.5 50 0
        16 1600 0
        32 3200 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.1343 0 -0.044
      rotation 0 -1 0 3.14159
      children [
        USE DIST_SENSOR
      ]
      name "rear bottom dist sonar"
      lookupTable [
        0.5 50 0
        16 1600 0
        32 3200 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.1309 0.016 -0.003
      rotation 0 0 1 1.5708
      children [
        USE DIST_SENSOR
      ]
      name "left side dist sonar"
      lookupTable [
        0.5 50 0
        10 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.1309 -0.016 -0.003
      rotation 0.7071067811864082 -0.7071067811864082 -6.276108057863702e-07 -3.1415853071795863
      children [
        USE DIST_SENSOR
      ]
      name "right side dist sonar"
      lookupTable [
        0.5 50 0
        10 1000 0
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.0255 0 -0.0519
      rotation 0.5773509358537164 -0.5773479358540498 -0.5773519358536053 -2.0943853071795866
      name "down front dist sonar"
      lookupTable [
        0.5 50 0
        11 1100 0
        22 2200 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.1123 0 -0.0519
      rotation 0.5773509358537164 -0.5773479358540498 -0.5773519358536053 -2.0943853071795866
      name "down back dist sonar"
      lookupTable [
        0.5 50 0
        11 1100 0
        22 2200 0.1
      ]
      type "sonar"
    }
    DistanceSensor {
      translation -0.026 0 0.02
      rotation 0 1 0 -1.5707953071795862
      name "top dist infrared"
      lookupTable [
        0.1 10 0
        8 800 0
      ]
      type "infra-red"
    }
  ]
  cameraSlot [
    Camera {
      rotation 0 1 0 0.3490659
      width 400
      height 240
    }
  ]
}
Robot {
  children [
    Receiver {
      name "StateReceiver"
      channel 4
    }
    Emitter {
      name "ActionEmitter"
      channel -1
    }
  ]
  name "agent"
  controller "<extern>"
  supervisor TRUE
}